src/server/conn/auto.rs

```diff
@@ -87,6 +87,40 @@ impl<E> Builder<E> {
         Ok(())
     }
 
+    /// Bind a connection together with a [`tower_service::Service`].
+    #[cfg(all(
+        any(feature = "http1", feature = "http2"),
+        any(feature = "server", feature = "client")
+    ))]
+    pub async fn serve_connection_tower<I, S, B>(&self, io: I, service: S) -> Result<()>
```

Comment:
Been thinking that axum also needs a version that uses `MakeService` for https://docs.rs/axum/latest/axum/struct.Router.html#method.into_make_service_with_connect_info